### PR TITLE
Split build jobs, add S3 push

### PR DESCRIPTION
**Pull request by pww217**
This splits build jobs into `unstable` - tracks main branch, tests all code on PR in pyscriptjs/

and `alpha` - which deploys a tagged/release build (convention tbd).

It also adds a step to configure AWS credentials (limited in scope to modify S3 bucket only) and push to the appropriate keys on that bucket. This uses an IAM role not full credentials to ensure that the runner (which is owned by Github) only has limited and temporary (short-lived token-based) access to AWS.

Anything that has to do with AWS auth is obscured or handled on the backend. I don't expect there should be any security concerns here. I designed it to be as secure as possible, but please do point out any flaws.

Updated: Last commit attempted to fix version pinning. Will test again.

### CODE REVIEW

**Review by fpliger — APPROVED** [review]
overall lgtm, thanks @pww217 ! I think @mattkram  has some reservations about linting in a different PR so I'll let him comment on that.

File: .github/workflows/sync-examples.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: '[CI] Sync Examples'
+
+on:
+    push: # Only run on merges into master that modify files under pyscriptjs/
```
Comment (mattkram) [inline .github/workflows/sync-examples.yml:4]:
```suggestion
    push: # Only run on merges into main that modify files under pyscriptjs/
```

**Review by mattkram — APPROVED** [review]
LGTM

File: .github/workflows/sync-examples.yml (hunk 1 above)
Comment (pww217) [reply .github/workflows/sync-examples.yml]: Thanks - force of habit from team edition.

**Comment by pww217** [comment]
Alright gonna merge and make sure these all work 🤞🏼 

It's not the live site right now so no harm if things go awry.

**Pull request merged**